src/content/blog/2020-12-07-roadmap-2021.md

```diff
@@ -0,0 +1,238 @@
+There are many types of externals ranging from globals over CommonJs/AMD/System to loading from classic script tag.
+Even `import()` (`type: "import"`) can be used to load an external, but `import` (`type: "module"`) can't be used yet.
+
+Interestingly even while `type: "module"` isn't supported yet, webpack already uses it as default when writing e. g. `import x from "https://example.com/module.js"`.
+The default has been choosing to seemlessly add support for ESM externals without introducing a breaking change.
+
+Absolute URLs in `import`s can make sense e. g. when using external services that offer their API as ESM: `import { event } from "https://analytics.company.com/api/v1.js"` (`import("https://analytics.company.com/api/v1.js")` might make more sense to gracefully handle errors when depending on this external service, but errors could also be caught higher in module graph).
+
+As usual the `externals` configuration allows to map any module name to externals:
+
+``` js
+export default {
+  externalsType: 'module',
+  externals: {
+    analytics: 'https://analytics.company.com/api/v1.js',
+    svelte: 'https://jspm.dev/svelte@3',
+    react: 'https://cdn.skypack.dev/preact@10',
+    'react-dom': 'https://esm.sh/[react,react-dom]/react-dom'
+  }
+};
+```
+
+W> Using multiple different ESM CDNs will not work. This is only an example.
+
+#### ESM library
+
+When ESM exports and import are supported people might think bundling a library makes sense, and that's probably true in some cases, but in many cases natively bundling will result in worse results.
+The biggest problem is the `"sideEffects": false` flag. It affects modules on per file base to skip whole modules. When concatenating multiple side-effect-free modules it's no longer possible to skip the individual modules, which leads to worse optimization when not all exports of the library are used.
+
+When the output should be a library that will be processed by a bundler later, this need to be considered.
+
+I could think of a special mode, which do no apply chunking and instead emit the raw (processed) modules connected via ESM imports and exports (or also CommonJS `require`).
+So this basically means loaders, module graph and asset optimizations run, but no chunk graph is created and each module in the module graph is emitted as separate file.
+
+#### Strict mode warnings
+
+When generating a ESM bundle, all contained code will be forced to strict mode.
+For many modules this isn't an issue, but there are a few older packages that might have trouble with the different semantic.
+We want to show warnings for these cases.
+
+### More first-class citizen
+
+webpack 4 and 5 did a lot work to support non-JS module types, and webpack 5 already supports some module types by default: JS (ESM/CJS/AMD), JSON, WebAssembly, Asset.
+Since webpack 5 one of our long term goals it to become a web-app optimizer, with the goal of supporting everything the browser supports.
```

Comment:
```suggestion
Since webpack 5 one of our long term goals is to become a web-app optimizer, with the goal of supporting everything the browser supports.
```